### PR TITLE
Test: Move client/lib/mixins/data-observe, client/lib/mixins/dirty-linked-state, client/lib/oauth-store tests to single test runner

### PR DESCRIPTION
**Pull request by gziolo**
Part of #3942 

Test folders moved:
* `client/lib/mixins/data-observe`
* `client/lib/mixins/dirty-linked-state`
* `client/lib/oauth-store`

### Testing
* `npm run test-client`
* `npm run test-client client/lib/mixins/data-observe/test/index.js  -- -w`
* `npm run test-client client/lib/mixins/dirty-linked-state/test/index.js  -- -w`
* `npm run test-client client/lib/oauth-store/test/index.js  -- -w`

### CODE REVIEW

File: client/lib/oauth-store/test/index.js [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
 	expect = require( 'chai' ).expect,
 	sinon = require( 'sinon' );
 
 /**
  * Internal dependencies
  */
-var oAuthToken = require( 'lib/oauth-token' ),
-	testConstants = require( '../constants' );
+import { actions } from '../constants';
+import * as oAuthToken from 'lib/oauth-token';
```
Comment (gziolo) [inline client/lib/oauth-store/test/index.js]: I had to import as object to use `sinon.stub`. I didn't find any way to stub single function...

File: client/lib/oauth-store/test/index.js [hunk 2]
```diff
@@ -88,16 +87,20 @@ describe( 'oAuthStore', function() {
...
-
-		global.document = { location: { replace: sinon.stub() } };
+	it( 'sets OAuth token when login is correct', sinon.test( function() {
+		const originalDocument = global.document,
+			replaceSpy = this.spy();
 
-		sinon.stub( oAuthToken, 'setToken' );
+		global.document = {
+			location: {
+				replace: replaceSpy
```
Comment (aduth) [inline client/lib/oauth-store/test/index.js]: Any particular reason this was changed from a stub to spy? The latter would typically be used when wanting to let an original behavior be left intact. Does JSDOM implement these functions for us? Could we simply `this.spy( global.document.location, 'replace' );` instead?

Comment (aduth) [inline client/lib/oauth-store/test/index.js]: Oh, we're not even pulling in `helpers/use-fake-dom` here. Maybe we should.

Comment (gziolo) [inline client/lib/oauth-store/test/index.js]: I need to experiment a bit more with that one. It was quite tricky to move it ;)

**Comment by aduth** [comment]
@gziolo : Was hoping to substitute shallow rendering in the dirty linked state tests. Turned out it didn't work, but had still migrated to Enzyme at that point, and I think it reads much better. Up to you if you want to keep d2d9f38.

File: client/lib/oauth-store/test/index.js (hunk 1 above)
Comment (gziolo) [inline client/lib/oauth-store/test/index.js]: Fixed, test that uses fake dom is a bit slower, but much more clean now :) Let me know if that makes sense.

**Comment by gziolo** [comment]
@aduth: thanks for migrating to Enzyme. Feels much better when you hide DOM element part :) I'll check details tomorrow.

File: client/lib/oauth-store/test/index.js (hunk 2 above)
Comment (gziolo) [inline client/lib/oauth-store/test/index.js]: Done :) It's a bit slower, but test file looks much clean in my opinion. Let me know if that works for you.

**Comment by aduth** [comment]
a1bec6735cee393ea697bee1e97ec8a71b95622e looks much improved. Good to merge from me :+1:

File: client/lib/mixins/dirty-linked-state/test/index.js [hunk 3]
```diff
@@ -2,9 +2,8 @@
  * External dependencies
  */
 import { expect } from 'chai';
-import ReactDom from 'react-dom';
+import { mount } from 'enzyme';
```
Comment (gziolo) [inline client/lib/mixins/dirty-linked-state/test/index.js]: This enzyme stuff looks mint! :100:

**Pull request merged**